test/cql-pytest/test_query_validation.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from cassandra.protocol import InvalidRequest
+from util import unique_name, unique_key_string
+
+
+@pytest.fixture(scope="module")
+def table1(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (id text, a int, b int, c list<int>, PRIMARY KEY((id),a,b))")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+
+@pytest.fixture(scope="module")
+def table2(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (id1 text, id2 text, a int, c list<int>, PRIMARY KEY((id1,id2),a))")
+    yield table
+    cql.execute("DROP TABLE " + table)
+
+
+def test_insert_with_list_column_and_missing_clustering_key_part(cql, table1):
+    key = unique_key_string()
+    with pytest.raises(InvalidRequest, match='Missing mandatory PRIMARY KEY part a'):
+        cql.execute(cql.prepare(f"INSERT INTO {table1} (id,b,c) VALUES ('{key}',1,null)"))
+
+
+def test_delete_with_list_column_and_missing_clustering_key_part(cql, table1):
+    key = unique_key_string()
+    cql.execute(cql.prepare(f"INSERT INTO {table1} (id,a,b,c) VALUES ('{key}',1,1,[1])"))
+    assert list(cql.execute(f"SELECT c FROM {table1} WHERE id='{key}' AND a=1 AND b=1"))[0][0] == [1]
+    with pytest.raises(InvalidRequest, match="Primary key column 'a' must be specified in order to modify column 'c'"):
+        cql.execute(cql.prepare(f"DELETE c FROM {table1} WHERE id='{key}' AND b=1"))
+    cql.execute(cql.prepare(f"DELETE c FROM {table1} WHERE id='{key}' AND a=1 AND b=1"))
+    assert list(cql.execute(f"SELECT c FROM {table1} WHERE id='{key}' AND a=1 AND b=1"))[0][0] is None
+
+
+def test_insert_with_list_column_and_missing_partition_key_part(cql, table2):
+    key = unique_key_string()
+    with pytest.raises(InvalidRequest, match='Missing mandatory PRIMARY KEY part id1'):
+        cql.execute(cql.prepare(f"INSERT INTO {table2} (id2,a,c) VALUES ('{key}',1,[1])"))
```

Comment:
Also here, since you already suspected we have problems with maybe missing the first component is different from missing the second component, that you also add in this test a check that an INSERT with the first component missing also results in the correct error.